### PR TITLE
Merge r1.7.1 to main

### PR DESCRIPTION
**Pull request by ericharper**
# What does this PR do ?

Add a one line overview of what this PR aims to accomplish.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 1]
```diff
@@ -236,12 +247,7 @@ def training_step(self, batch, batch_idx):
 
         # we zero grads here because we also call backward in the apex fwd/bwd functions
         self._optimizer.zero_grad()
-
-        if self.use_soft_prompts:
```
Comment (vadam5) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:240]: I don't think this if/else statement should be removed

**Review by vadam5 — CHANGES_REQUESTED** [review]
Looks good other than my one comment in `megatron_gpt_model.py`

File: nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py [hunk 2]
```diff
@@ -236,12 +247,7 @@ def training_step(self, batch, batch_idx):
...
-            # The micro batches are already prepared for apex by the prompt tuning dataclass
-            batch_for_pipeline = batch
-            tensor_shape = [len(batch_for_pipeline[0][0]), self.cfg.micro_batch_size, self.cfg.hidden_size]
-        else:
+>
```
Comment (vadam5) [inline nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py:250]: Also this should be deleted

**Review by MaximumEntropy — DISMISSED** [review]
LGTM!

**Review by vadam5 — APPROVED** [review]
LGTM!

**Pull request merged**